health/health_json.c

```diff
@@ -93,18 +93,19 @@ inline void health_alarm_entry2json_nolock(BUFFER *wb, ALARM_ENTRY *ae, RRDHOST
     buffer_strcat(wb, "\t}");
 }
 
-void health_alarm_log2json(RRDHOST *host, BUFFER *wb, uint32_t after) {
+void health_alarm_log2json(RRDHOST *host, BUFFER *wb, uint32_t after, char *chart) {
     netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);
 
     buffer_strcat(wb, "[");
 
     unsigned int max = host->health_log.max;
     unsigned int count = 0;
     ALARM_ENTRY *ae;
-    for(ae = host->health_log.alarms; ae && count < max ; count++, ae = ae->next) {
-        if(ae->unique_id > after) {
+    for(ae = host->health_log.alarms; ae && count < max ; ae = ae->next) {
+        if( (ae->unique_id > after && !chart) || (ae->unique_id > after && chart && !strcmp(chart, ae->chart)) ) {
```

Comment:
Calculate the chart hash and try to use the `hash_chart` that exists in the alarm_entry

Reply:
Hi Stelio, thanks, I've pushed: https://github.com/netdata/netdata/pull/10760/commits/07c231f9c6e08746f65f39915756b4c013fbdfbf for this.